userdata/userdata.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Steffen Vogel.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package userdata implements a TLV parser/serializer for libnftables-compatible comments
+package userdata
+
+import (
+	"bytes"
+	"encoding/binary"
+)
+
+type Type byte
+
+const (
+	TypeComment Type = iota
+	TypeRuleID  Type = 100 // custom extension
```

Comment:
Where is this defined upstream? Can you add a link?

Reply:
I've added a link as well as a second type constant of which I was not aware yet.